cesium/internal/domain/file_controller.go

```diff
@@ -200,19 +207,56 @@ func (fc *fileController) gcReaders() (bool, error) {
 	return false, nil
 }
 
-func (fc *fileController) gcWriters() (bool, error) {
+func (fc *fileController) atDescriptorLimit() bool {
 	fc.writers.RLock()
-	for i, w := range fc.writers.open {
-		if w.tryAcquire() {
-			fc.writers.RUnlock()
-			fc.writers.Lock()
-			err := w.TrackedWriteCloser.Close()
-			fc.writers.open = append(fc.writers.open[:i], fc.writers.open[i+1:]...)
-			fc.writers.Unlock()
-			return true, err
+	fc.readers.RLock()
+	defer func() {
+		fc.writers.RUnlock()
+		fc.readers.RUnlock()
```

Comment:
I would generally release locks in the reverse order you acquired them because it lets you better ensure a deadlock won't occur (I can explain more since it might not be immediately clear why)